### DISCUSSION TITLE
Reynolds number in lock-exchange and axisymmetric gravity current case

### DISCUSSION THREAD

**Discussion by jianzhou722** [xcompact3d/Incompact3d#106] [Q&A] [answered]
Hi everyone,

I am tring to run the lock-exchange and axisymmetric gravity current tutorial cases. I cannot find the input parameters for the density of the lock and the ambient fluid. Is it possible that this can be set via the Reynolds number, Froude number, or any other parameters? What's the role of these numbers in the simulation?

Also, all the parameters are dimensionless. Is it correct that I can consider all of them to be in SI units? For example, in axisymmetric gravity current case, the domain size is 20*2*20, does this mean 20 m*2 m*20 m, or 20 cm*2 cm*20 cm? Time step is 0.0005, is this in seconds?

Thanks in advance for your help!
Jay

**Comment by slaizet** [comment]
Hi Jack,

The code is designed for non-dimensional simulations, except possibly for the wind farm simulator. For the lock-exchange case (in the lock exchange example directory), the data are normalised with the buoyancy velocity, the heavy density and a length scale H, which corresponds to the height of the channel. See [Bartholomew, P., & Laizet, S. (2019). A new highly scalable, high-order accurate framework for variable-density flows: Application to non-Boussinesq gravity currents. Computer Physics Communications, 242, 83-94] for more details. 

In general, you can decide to do non-dimensional or dimensional simulations, as long as your are consistent with all the parameters. 

I am not quite sure what you are talking about regarding the axisymmetric gravity current case, as it is not available in the master branch of the code.

Best,

Sylvain

**Reply by jianzhou722** [reply]
Thank you! When I turn off the LMN model, the densities of the two liquids (dens1 and dens2) won't be read by input.i3d. In this case, how can I set the density difference between the two liquids?

In the non-LMN case, is the gravity current instead driven by the scalar concentration of 1 in the lock versus 0 in the ambient? I am confused...

**Reply by jianzhou722** [reply]
Would you also please give me some hint on how to set the value of Reynolds number in the lock-exchange case? In input.i3d, it says nu=1/re. According to this, I tried to set re = 1/1e-6 = 1e6, where 1e-6 is the kinematic viscosity of water. However, the scalar always diverge in the simulation.

Thank you!

**Comment by slaizet** [comment, marked as answer]
Different authors have used different Reynolds numbers. As far as we are concerned, you will find the definition of our Reynolds number in [Espath, L. F. R., Pinto, L. C., Laizet, S., & Silvestrini, J. H. (2015). High-fidelity simulations of the lobe-and-cleft structures and the deposition map in particle-driven gravity currents. Physics of Fluids, 27(5), 056604]. Basically, it is based on the buoyancy velocity and the height of the domain (which are used to make all the quantities dimensionless, hence why Re=1/nu).

As I said before, all the quantities in the code have no dimension, so you cannot set up the Reynolds number based on the kinematic viscosity of water (not to mention that you need to pay attention to the numerical stability of the code by adapting time step and spatial resolution).

If the LMN model is off, then the flow is fully incompressible with density=1. The particle concentration is then simulated via the Boussinesq approximation.

I suggest you read carefully our papers on gravity currents so you can understand how the parameters are defined and how all the quantities are made dimensionless.

Best,

Sylvain
